pageserver/src/tenant/mgr.rs

```diff
@@ -336,10 +336,20 @@ pub async fn delete_timeline(
     Ok(())
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum TenantStateError {
+    #[error("tenant is not found")]
+    NotFound(TenantId),
+    #[error("tenant is stopping")]
+    IsStopping(TenantId),
+    #[error("tenant is broken")]
+    Broken(TenantId),
+}
+
 pub async fn detach_tenant(
     conf: &'static PageServerConf,
     tenant_id: TenantId,
-) -> anyhow::Result<()> {
+) -> anyhow::Result<(), TenantStateError> {
```

Comment:
anyhow::Result<T, E> is the same as `Result<T, E>`, let's use the latter since it's less confusing.